spec/requests/api/generic_events_controller_create_spec.rb

```diff
@@ -47,8 +42,42 @@
       before { do_post }
     end
 
+    context 'when event-specific relationships are in the details attribute' do
```

Comment:
so I understand that we want to merge relationships into details for reporting purposes, which is ok. But I don't think this should pass, we shouldn't allow users to set relationships through details attribute by mistake, due to the previously mentioned note around JSON:API compliance.